jormungandr/src/start_up/mod.rs

```diff
@@ -63,12 +64,14 @@ pub async fn prepare_block_0(
             })
         }
         Block0Info::Hash(block0_id) => {
-            if storage.block_exists(&block0_id)? {
+            let connection = storage.connect().unwrap();
```

Comment:
This is a blocking APl, should not be used in async code without offloading the work to the threadpool scheduler?

Reply:
This can be problematic. Using the threadpool scheduler requires `'static`, the signature of this function is `storage.connect(&self)` and `storage` is not `'static`. We can wrap the `storage` into `Arc` like we do in `Storage03` and always pass it around like that.

Reply:
Let's make sure we are on the the same page with this. I think the ideal solution is to have a thread/DB connection pool specifically for synchronous storage operations, with an API that would allow Tokio 0.2 tasks to submit closures of the form `|connection| { /* do things synchronously with the storage connection */ }` and await the results asynchronously. The connections are created or recycled by the pool implementation and passed to the closure.